src/setup/rgw.py

```diff
@@ -538,13 +524,30 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf, data_pod_hostname: s
         if rgw_lock is True:
             Log.info('Creating admin user.')
             # Before creating user check if user is already created.
-            status = Rgw._create_rgw_user(conf)
-            if status == 0:
+            user_status = Rgw._create_rgw_user(conf)
+
+            if user_status != 0:
+                current_data_node = Rgw._get_current_data_node()
+                machine_ids = Rgw._get_cortx_conf(conf, 'cluster>storage_set[0]>nodes')
+                data_pod_hostnames = [Rgw._get_cortx_conf(conf, \
+                    f'node>{machine_id}>hostname') for machine_id in machine_ids if \
+                    Rgw._get_cortx_conf(conf, f'node>{machine_id}>type') == 'data_node']
+                data_pod_hostnames.remove(current_data_node)
+                for data_pod_hostname in data_pod_hostnames:
+                    status = Rgw._update_hax_endpoint_and_create_admin(conf, \
```

Comment:
why this recursive call? Lock is there, we can just loop around the required logic !!

Reply:
Can be done in 2 ways.
1. recursive call. (Reads other data pods information only if current data pod fails.)
2. Read all data_pods and iterate till it returns success.

Reply:
Done.